### PR TITLE
Drop `extend` CJS dependency

### PR DESCRIPTION
**Pull request by fuma-nama**
### Initial checklist

* [x] I read the support docs 
* [x] I read the contributing guide 
* [x] I agree to follow the code of conduct 
* [x] I searched issues and discussions and couldn’t find anything or linked relevant results below 
* [x] I made sure the docs are up to date
* [x] I included tests (or that’s not needed)

### Description of changes

Removed the remaining CJS dependency `extend`, since unified itself is ESM-only, it would be great if all deps are also ESM so it doesn't need bundling to work in browser environments.

The `extend` package is being used as a deep merge utility, while its core functionality is being replaced by `Object.assign()`. I think a proper deep merge utility would do better, but `extend` has some behaviours that may differs from existing deep merge utilities, so I rewrote the original utility with modern ESM, shouldn't cause any breaking changes.

I had to lower the test coverage as some edge cases are not easily reached, let me know if this isn't desirable.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -87,7 +85,7 @@
     "prepack": "npm run build && npm run format",
     "test": "npm run build && npm run format && npm run test-coverage",
     "test-api": "node --conditions development test/index.js",
-    "test-coverage": "c8 --100 --check-coverage --reporter lcov npm run test-api"
+    "test-coverage": "c8 --96 --check-coverage --reporter lcov npm run test-api"
```
Comment (Murderlon) [inline package.json]: This is not going to be acceptable for some I think. But from my side it's fine if the code is basically as-is from the `extend` package and we're never going to touch it.

File: lib/deep-merge.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+/**
+ * Deeply merge plain objects and arrays.
+ *
+ * @template {Record<string, unknown> | Array<unknown>} Target
+ * @param {Target} target
+ * @param {unknown} source
+ * @returns {Target}
+ */
+export function deepMerge(target, source) {
```
Comment (Murderlon) [inline lib/deep-merge.js:37]: Did you literally copy the contents of `extend`? I would prefer and trust that more than some AI attempt I think 🤔

**Review by Murderlon — COMMENTED** [review]
Not up to me to decide whether this goes in. Knowing @wooorm, he really likes consistency. So that usually means using `extend` everywhere or nowhere, not just dropping in one package. Luckily it seems there are [only two](https://github.com/search?q=%2F%22extend%22%2F+path%3Apackage.json+%28org%3Aremarkjs+OR+org%3Arehypejs+OR+org%3Aretextjs+OR+org%3Aunifiedjs+OR+org%3Asyntax-tree+OR+org%3Avfile%29&type=code&query=%28%22extend%22+path%3Apackage.json%29+%28org%3Aremarkjs+OR+org%3Arehypejs+OR+org%3Aretextjs+OR+org%3Aunifiedjs+OR+org%3Asyntax-tree+OR+org%3Avfile%29)

File: lib/deep-merge.js (hunk 2 above)
Comment (fuma-nama) [reply lib/deep-merge.js:37]: yeah I was writing it side-by-side with the original implementation of `extend`

File: package.json (hunk 1 above)
Comment (fuma-nama) [reply package.json]: perhaps I can add more unit tests to hit the coverage, will take a look later

**Comment by fuma-nama** [comment]
Oops I didn't search in other repositories, I think that will need another PR, would like to hear some inputs from maintainers before doing that. `extend` doesn't actually merge data structures like Map & Set, there's better alternatives for deep merge utils.

**Comment by Murderlon** [comment]
Taking a step back, wondering if we can use `structuredClone`, [browser support](https://caniuse.com/wf-structured-clone) is good, also in Node.js since 17.0.0 🤔

**Comment by fuma-nama** [comment]
I think some libraries might store functions in the `settings` object, I would prefer to not break anything when possible.

File: package.json (hunk 1 above)
Comment (fuma-nama) [reply package.json]: added more unit tests, back to 100% coverage now.

**Comment by wooorm** [comment]
Sorry but I don‘t think this is possible.

I tried structuredClone. It passed tests. It worked in several of the 100s of repos. Then in failed in some. Some things put functions on data.
There are lots of people writing their own plugins and doing all kinds of things.

This requires lots and lots of tests to have parity with `extend`.
At that point, put it in a new repo, share it with the world. Maybe then it can be used here.

See GH-263, GH-153
